src/lib/blocks.js

```diff
@@ -32,23 +32,100 @@ module.exports = function (vm) {
     const backdropsMenu = function () {
         return vm.runtime.targets[0].sprite.costumes.map(costume => [costume.name, costume.name]);
     };
+    
+    const spriteMenu = function () {
+        var i = 0;
+        var sprites = [];
+        var k = 0;
+        for (i = 0; i < vm.runtime.targets.length; i++) {
+            if (vm.runtime.targets[i].isOriginal === true) {
+                if (vm.runtime.targets[i].isStage === false) {
+                    if (vm.runtime.targets[i].sprite.name === vm.editingTarget.sprite.name) {
```

Comment:
It would be safer to use the sprite `id` rather than its `name`, since that is guaranteed to be unique.